autobot-backend/services/agent_org_service.py

```
@@ -293,6 +330,23 @@ async def upsert_node(
+
+        # Tenant scoping (#15794). Same reasoning as update_reporting_line: the
+        # permission says the caller may edit reporting lines, not whose. An
+        # existing node belonging to another company must read as absent rather
+        # than be updated, and a manager must be in this company.
+        #
+        # Absent, not forbidden: a distinct error would tell an unauthorised
+        # caller that an agent exists in a company they cannot see.
+        company_uuid = _normalise_company_id(company_id)
+        if company_uuid is not None:
+            if node is not None and node.company_id != company_uuid:
+                raise ValueError(f"Agent not found in org hierarchy: {agent_id!r}")
```

Ruff (0.16.3)

[warning] Avoid specifying long messages outside the exception class

(TRY003)